x-pack/test/security_solution_api_integration/test_suites/detections_response/detection_engine/rule_execution_logic/trial_license_complete_tier/execution_logic/eql_alert_suppression.ts

```diff
@@ -1786,40 +1793,1230 @@ export default ({ getService }: FtrProviderContext) => {
+        expect(sequenceAlert.length).toEqual(1);
+
+        // suppression start equal to alert timestamp
+        const suppressionStart = sequenceAlert[0]._source?.[TIMESTAMP];
+
+        expect(sequenceAlert[0]._source).toEqual({
+          ...sequenceAlert[0]._source,
+          [ALERT_SUPPRESSION_TERMS]: [
+            {
+              field: 'host.name',
+              value: ['host-a'],
+            },
+          ],
+          // suppression boundaries equal to original event time, since no alert been suppressed
+          [ALERT_SUPPRESSION_START]: firstTimestamp,
+          [ALERT_SUPPRESSION_END]: firstTimestamp,
+          [TIMESTAMP]: suppressionStart,
+          [ALERT_SUPPRESSION_DOCS_COUNT]: 0,
+        });
+
+        const secondTimestamp = new Date(Date.now() - 1000).toISOString();
+        const secondTimestamp2 = new Date(Date.now()).toISOString();
+        const secondDocument = {
+          id,
+          '@timestamp': secondTimestamp,
+          host: {
+            name: 'host-a',
+          },
+        };
+
+        // Add a new document, then disable and re-enable to trigger another rule run.
+        // the second rule run should generate a new alert
+        await patchRule(supertest, log, { id: createdRule.id, enabled: false });
+
+        await indexListOfSourceDocuments([
+          secondDocument,
+          { ...secondDocument, '@timestamp': secondTimestamp2 },
+        ]);
+        await patchRule(supertest, log, { id: createdRule.id, enabled: true });
+        const afterTimestamp = new Date(Date.now() + 1000);
+        const secondAlerts = await getOpenAlerts(
+          supertest,
+          log,
+          es,
+          createdRule,
+          RuleExecutionStatusEnum.succeeded,
+          undefined,
+          afterTimestamp
+        );
+
+        const [sequenceAlert2] = partition(
+          secondAlerts.hits.hits,
+          (alert) => alert?._source?.['kibana.alert.building_block_type'] == null
+        );
+
+        expect(sequenceAlert2.length).toEqual(2);
+        expect(sequenceAlert2[0]._source).toEqual({
+          ...sequenceAlert2[0]?._source,
+          [ALERT_SUPPRESSION_TERMS]: [
+            {
+              field: 'host.name',
+              value: ['host-a'],
+            },
+          ],
+          [ALERT_SUPPRESSION_START]: firstTimestamp,
+          [ALERT_SUPPRESSION_END]: firstTimestamp,
+          [ALERT_SUPPRESSION_DOCS_COUNT]: 0, // 1 alert from second rule run, that's why 1 suppressed
+        });
+        expect(sequenceAlert2[1]._source).toEqual({
+          ...sequenceAlert2[1]?._source,
+          [ALERT_SUPPRESSION_TERMS]: [
+            {
+              field: 'host.name',
+              value: ['host-a'],
+            },
+          ],
+          [ALERT_SUPPRESSION_START]: secondTimestamp2,
+          [ALERT_SUPPRESSION_END]: secondTimestamp2,
+          [ALERT_SUPPRESSION_DOCS_COUNT]: 0, // 1 alert from second rule run, that's why 1 suppressed
+        });
       });
```

Comment:
can we add one test for a query with  3+ sequences? So we are sure, this part works too